File: 231.power-of-two.js
/*
 * @lc app=leetcode id=231 lang=javascript
 *
 * [231] Power of Two
 */

// @lc code=start
/**
 * @param {number} n
 * @return {boolean}
 */
var isPowerOfTwo = function (n) {
    if (n === 1) {
        return true;
    }
    if (n < 1) {
        return false;
    }
    while (n > 1) {
        if (n % 2 === 0) {
            n = n / 2;
        } else {
            return false;
        }
    }
    return true;
};
// @lc code=end

File: 231.power-of-two.py
#
# @lc app=leetcode id=231 lang=python
#
# [231] Power of Two
#

# @lc code=start
class Solution(object):
    def isPowerOfTwo(self, n):
        """
        :type n: int
        :rtype: bool
        """

        if n == 0:
            return False

        while n % 2 == 0:
            n /= 2

        return n == 1

# @lc code=end
